### DISCUSSION TITLE
Doesn't work with AWS Lambdas

### DISCUSSION THREAD

**Discussion by jovi-tsx** [ai/nanoid#451] [General]
Greetings! I'm currently having an issue with nanoid at my serverless stack backend + lambda function.

```ts
import { customAlphabet } from "nanoid";

export const generateNid = () => {
  const nanoid = customAlphabet(
    "1234567890ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz",
    10
  );

  const nid = nanoid();

  console.log(nid);

  return nanoid() + new Date().getFullYear();
};

export const create = lambda(async (event) => {
  const body = event.body as string;
  const sale: Sale = JSON.parse(body);

  let nid = "SA" + generateNid();
  let nidExists = !!(await SaleModel.scan("nid").eq(nid).exec()).count;

  while (nidExists) {
    nid = "SA" + generateNid();
    nidExists = !!(await SaleModel.scan("nid").eq(nid).exec()).count;
  }

  return SaleModel.create({
    id: uuid.v1(),
    nid,
    ...sale
  });
});
```

This returns sometimes that "crypto.getRandomValues() is not a function". Other times, it will work, but it will generate only "1111111111"

**Comment by ai** [comment]
You need to have Web Crypto API to use Nano ID.

We can change API if you tell us how to fix it. But I am not AWS user to give you my solution.

**Comment by monestry-dev** [comment]
If I might chime in, Web Crypto API has only been available since Node 15, so do check if your lambda's Node version is 15 or later.

**Comment by jovi-tsx** [comment]
My env is Node 16, so this is not the problem. I am using own nodejs crypto module and it's working, if i remember correctly this error is due to the fact that getRandomValues doesn't work on server side. I used randomBytes with a custom logic for my case, but if anyone else find yourself with the same problem, maybe this comment can help. To solve this, however, I can't give you guys a proper solution since I am not an AWS expert myself
